src/Common/src/System/Security/Cryptography/ECDsaSecurityTransforms.cs

```diff
@@ -0,0 +1,542 @@
+                        SetKey(SecKeyPair.PublicOnly(publicKey));
+                    }
+                }
+
+                public override void GenerateKey(ECCurve curve)
+                {
+                    curve.Validate();
+
+                    if (!curve.IsNamed)
+                    {
+                        throw new PlatformNotSupportedException(SR.Cryptography_ECC_NamedCurvesOnly);
+                    }
+
+                    int keySize;
+
+                    switch (curve.Oid.Value)
+                    {
+                        // secp256r1 / nistp256
+                        case "1.2.840.10045.3.1.7":
+                            keySize = 256;
+                            break;
+                        // secp384r1 / nistp384
+                        case "1.3.132.0.34":
+                            keySize = 384;
+                            break;
+                        // secp521r1 / nistp521
+                        case "1.3.132.0.35":
+                            keySize = 521;
+                            break;
+                        default:
+                            throw new PlatformNotSupportedException(
+                                SR.Format(SR.Cryptography_CurveNotSupported, curve.Oid.Value));
+                    }
+
+                    // Clear the current key, because GenerateKey on the same curve makes a new key,
+                    // unlike setting the KeySize property to the current value.
+                    SetKey(null);
+                    KeySizeValue = keySize;
+
+                    // Generate the keys immediately, because that's what the verb of this method is.
+                    GetKeys();
+                }
+
+                private static SafeSecKeyRefHandle ImportKey(ECParameters parameters)
+                {
+                    bool isPrivateKey = parameters.D != null;
+                    byte[] blob = isPrivateKey ? parameters.ToPrivateKeyBlob() : parameters.ToSubjectPublicKeyInfo();
+
+                    return Interop.AppleCrypto.ImportEphemeralKey(blob, isPrivateKey);
+                }
+
+                private void SetKey(SecKeyPair newKeyPair)
+                {
+                    SecKeyPair current = _keys;
+                    _keys = newKeyPair;
+                    current?.Dispose();
+
+                    if (newKeyPair != null)
+                    {
+                        long size = Interop.AppleCrypto.EccGetKeySizeInBits(newKeyPair.PublicKey);
+
+                        Debug.Assert(size == 256 || size == 384 || size == 512, $"Unknown keysize ({size})");
+                        KeySizeValue = (int)size;
+                    }
+                }
+
+                private SecKeyPair GetKeys()
+                {
+                    SecKeyPair current = _keys;
+
+                    if (current != null)
+                    {
+                        return current;
+                    }
+
+#if APPLE_ASYMMETRIC_GENERATION
```

Comment:
It's from the 3rd commit: Temporarily disable key generation.  I expect to undo that whole commit once the keychain pollution problem is solved; but since it's not necessary for starting X.509 work I didn't want to make it a blocker.